shared/search/promises.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Evaluate each promise sequentially. Return the result of the promise
+ * to yield a non-empty set of results. The result of subsequent promises
+ * will be ignored.
+ *
+ * @param ps A list of in-flight promises.
+ */
+export async function getFirst<T>(...ps: Promise<T[]>[]): Promise<T[]> {
```

Comment:
The comment update helps, thanks. 

I still think this shouldn't be exported as a library function. We only need it for one very specific case. I think it should be unexported and made part of that library, until we discover we need it for something else. It could be simplified too, since we only ever care about two:

```typescript
async function getFirst<T>(p: Promise<T[]>, q: Promise<T[]>): Promise<T[]> {
   const values = await p
   if (values.length !== 0) {
      return values
   }
   return await q
}
```

or words to that effect.

This isn't a blocker, though.

Reply:
FYI, you cant test it if you can't import it.

And I don't see how a variadic function is much more complex than something that takes exactly two, especially if the body of the function is going to treat them as a sequential list. I wrote this function specifically this way because I can see us possibly needing to wait on a third value if we need to change the fallback behavior in the near future.

Reply:
I'd simply like to avoid unnecessary premature abstraction. With this simplification, you wouldn't arguably even need a test.